crates/system/interactors/src/testing/test_spot_check_interactor.rs

```diff
@@ -0,0 +1,45 @@
+use crate::prelude::*;
+
+pub struct TestSpotCheckInteractor {
+    user: SpotCheckUser,
+}
+
+#[async_trait::async_trait]
+impl SpotCheckInteractor for TestSpotCheckInteractor {
+    async fn spot_check(
+        &self,
+        _factor_source: FactorSource,
+        _allow_skip: bool,
+    ) -> Result<SpotCheckResponse> {
+        match self.user.clone() {
+            SpotCheckUser::Failure(common_error) => Err(common_error),
+            SpotCheckUser::Valid => Ok(SpotCheckResponse::Valid),
+            SpotCheckUser::Skipped => Ok(SpotCheckResponse::Skipped),
+        }
+    }
+}
+
+#[derive(Clone)]
+pub enum SpotCheckUser {
+    Failure(CommonError),
```

Comment:
`Skipped` is past tense of a **verb**, `Valid` and `Failure` are not verbs. Pick a part of speech kind and use for all variants? :)